astroquery/solarsystem/pds/core.py

```diff
@@ -0,0 +1,493 @@
+                (
+                    "extra_ra",
+                    "",
+                ),  # figure options below this line, can all be hardcoded and ignored
+                ("extra_ra_type", "hours"),
+                ("extra_dec", ""),
+                ("extra_name", ""),
+                ("title", ""),
+                ("labels", "Small (6 points)"),
+                ("moonpts", "0"),
+                ("blank", "No"),
+                ("opacity", "Transparent"),
+                ("peris", "None"),
+                ("peripts", "4"),
+                ("arcpts", "4"),
+                ("meridians", "Yes"),
+                ("output", "html"),
+            ]
+        )
+
+        # return request_payload if desired
+        if get_query_payload:
+            return request_payload
+
+        # set return_raw flag, if raw response desired
+        if get_raw_response:
+            self.return_raw = True
+
+        # query and parse
+        response = self._request(
+            "GET", URL, params=request_payload, timeout=self.TIMEOUT, cache=cache
+        )
+        self.uri = response.url
+
+        return response
+
+    def _parse_ringnode(self, src):
+        """
+        Routine for parsing data from ring node
+
+        Parameters
+        ----------
+        self : RingNodeClass instance
+        src : list
+            raw response from server
+
+
+        Returns
+        -------
+        systemtable : dict
+        bodytable : `astropy.Table`
+        ringtable : `astropy.Table`
+        """
+
+        self.raw_response = src
+        soup = BeautifulSoup(src, "html.parser")
+        text = soup.get_text()
+        # print(repr(text))
+        textgroups = re.split("\n\n|\n \n", text)
+        ringtable = None
+        for group in textgroups:
+            group = group.strip(", \n")
+
+            # input parameters. only thing needed is obs_time
+            if group.startswith("Observation"):
+                obs_time = group.split("\n")[0].split("e: ")[-1].strip(", \n")
+
+            # minor body table part 1
+            elif group.startswith("Body"):
+                group = "NAIF " + group  # fixing lack of header for NAIF ID
+                bodytable = ascii.read(
+                    group,
+                    format="fixed_width",
+                    col_starts=(0, 4, 18, 35, 54, 68, 80, 91),
+                    col_ends=(4, 18, 35, 54, 68, 80, 91, 102),
+                    names=(
+                        "NAIF ID",
+                        "Body",
+                        "RA",
+                        "Dec",
+                        "RA (deg)",
+                        "Dec (deg)",
+                        "dRA",
+                        "dDec",
+                    ),
+                )
+                units_list = [None, None, None, None, u.deg, u.deg, u.arcsec, u.arcsec]
+                bodytable = table.QTable(bodytable, units=units_list)
```

Comment:
I know this is not consistent in astroquery, but you could initiatilize the table directly. I think it's much more legible, and also doesn't create another Table instance:

```
bodytable = QTable.read(group, format="ascii.fixed_width",
                        col_starts=(0, 4, 18, 35, 54, 68, 80, 91),
                        col_ends=(4, 18, 35, 54, 68, 80, 91, 102),
                        names=("NAIF ID", "Body", "RA", "Dec", "RA (deg)", "Dec (deg)", "dRA", "dDec"),
                        units=([None, None, None, None, u.deg, u.deg, u.arcsec, u.arcsec]))
```

Reply:
same logic applies to all other QTables or Tables below

Reply:
thank you for this! I was looking for a way to do this, and did not realize that you could pass ascii.fixed_width to the format keyword

Reply:
All the early modules parsed the ascii just as you did, so I suspect if we dig into the history there might have been some Table.read() limitations in those times. Basically with `Table.read()` you call the ready of the chosen format, and then have access all its options.